app/src/org/commcare/CommCareHeartbeatManager.java

```diff
@@ -0,0 +1,309 @@
+
+        @Override
+        public void processSuccess(int responseCode, InputStream responseData) {
+            try {
+                String responseAsString = StreamsUtil.inputStreamToByteArray(responseData).toString();
+                JSONObject jsonResponse = new JSONObject(responseAsString);
+                parseHeartbeatResponse(jsonResponse);
+            }
+            catch (JSONException e) {
+                Logger.log(AndroidLogger.TYPE_ERROR_SERVER_COMMS,
+                        "Heartbeat response was not properly-formed JSON: " + e.getMessage());
+            }
+            catch (IOException e) {
+                Logger.log(AndroidLogger.TYPE_ERROR_SERVER_COMMS,
+                        "IO error while processing heartbeat response: " + e.getMessage());
+            }
+        }
+
+        @Override
+        public void processRedirection(int responseCode) {
+            processErrorResponse(responseCode);
+        }
+
+        @Override
+        public void processClientError(int responseCode) {
+            processErrorResponse(responseCode);
+        }
+
+        @Override
+        public void processServerError(int responseCode) {
+            processErrorResponse(responseCode);
+        }
+
+        @Override
+        public void processOther(int responseCode) {
+            processErrorResponse(responseCode);
+        }
+
+        @Override
+        public void handleIOException(IOException exception) {
+            Logger.log(AndroidLogger.TYPE_ERROR_SERVER_COMMS,
+                    "Encountered IOException while getting response stream for heartbeat response: "
+                            + exception.getMessage());
+        }
+
+        private void processErrorResponse(int responseCode) {
+            Logger.log(AndroidLogger.TYPE_ERROR_SERVER_COMMS,
+                    "Received error response from heartbeat request: " + responseCode);
+        }
+    };
+
+    private static CommCareHeartbeatManager INSTANCE;
+    public static CommCareHeartbeatManager instance() {
+        if (INSTANCE == null) {
+            INSTANCE = new CommCareHeartbeatManager();
+        }
+        return INSTANCE;
+    }
+
+    public void startHeartbeatCommunications() {
+        TimerTask heartbeatTimerTask = new TimerTask() {
+            @Override
+            public void run() {
+                try {
+                    User currentUser = CommCareApplication.instance().getSession().getLoggedInUser();
+                    //simulateRequestGettingStuck();
+                    //requestHeartbeat(currentUser);
+                    parseTestHeartbeatResponse();
+                } catch (SessionUnavailableException e) {
+                    // Means the session has ended, so we should stop these requests
+                    stopHeartbeatCommunications();
+                }
+            }
+        };
+
+        heartbeatTimer = new Timer();
+        heartbeatTimer.schedule(heartbeatTimerTask, new Date(), ONE_DAY_IN_MS);
```

Comment:
Hey, I'm a little confused about this comment. In the minimal discussions we had about how to set this up, you had explicitly said that you weren't sure whether it was preferable to tie the heartbeat's lifecycle to being triggered post-form submissions, vs. just set it up to happen on a timer, and that I should go with whichever choice ended up seeming easier/better to me. We also then later discussed that I was planning on using the timer strategy because I mentioned to you that I had found discussions online where people were wary of using a `TimerTask`, and you checked those out and then we agreed that it seemed like the concerns weren't relevant to us, and you agreed that that seemed like a good option. So I'm confused why now you're surprised that I went with the timer option over the tied-to-form-submission option.